scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -181,15 +188,54 @@ def _highly_variable_genes_seurat_v3(
         return df
 
 
+@dataclass
+class _Cutoffs:
+    min_disp: float
+    max_disp: float
+    min_mean: float
+    max_mean: float
+
+    @classmethod
+    def validate(
+        cls,
+        *,
+        n_top_genes: int | None,
+        min_disp: float,
+        max_disp: float,
+        min_mean: float,
+        max_mean: float,
+    ) -> _Cutoffs | int:
+        if n_top_genes is None:
+            return cls(min_disp, max_disp, min_mean, max_mean)
+
+        cutoffs = {"min_disp", "max_disp", "min_mean", "max_mean"}
+        defaults = {
+            p.name: p.default
+            for p in signature(highly_variable_genes).parameters.values()
+            if p.name in cutoffs
+        }
+        if {k: v for k, v in locals().items() if k in cutoffs} != defaults:
+            logg.info("If you pass `n_top_genes`, all cutoffs are ignored.")
+        return n_top_genes
+
+    def in_bounds(
+        self,
+        mean: NDArray[np.float64] | DaskArray,
+        dispersion_norm: NDArray[np.float64] | DaskArray,
```

Comment:
```suggestion
        mean: NDArray[np.floating] | DaskArray,
        dispersion_norm: NDArray[np.floating] | DaskArray,
```

This would work on a broader range of types, no?

Reply:
true. In practice, there are sometimes problems with numpy’s overloading when one uses `floating` instead of concrete types, but here it should be fine.